src/listener.js

```diff
@@ -23,23 +23,32 @@ const sioOptions = {
 module.exports = ({ handler, upgrader }, WebRTCStar, options = {}) => {
   const listener = new EventEmitter()
   let listeningAddr
+  let sioUrl
 
   listener.__connections = []
   listener.__spChannels = new Map()
   listener.__pendingIntents = new Map()
   listener.listen = (ma) => {
+    // Should only be used if not already listening
+    if (listeningAddr) {
+      throw errCode(new Error('listener already in use'), 'ERR_ALREADY_LISTENING')
+    }
+
     const defer = pDefer()
 
+    // Should be kept unmodified
     listeningAddr = ma
+
+    let signallingAddr
```

Comment:
I think it would be a lot easier to comprehend this code if this was

```suggestion
    const signallingAddr = withPeerId(ma, upgrader.localPeer)
```

where `withPeerId` would encapsulate logic below.

P.S. This is a third place I see this logic in last two days, makes me wonder if it should be just part of `ma` itself.

Reply:
Perhaps, it would be a good idea! I will merge this as it is, given that this was already as it is. 
Will keep a note for next time I get to times code